includes/data-port/class-sensei-import-csv-reader.php

```diff
@@ -60,21 +60,60 @@ class Sensei_Import_CSV_Reader {
 	public function __construct( $csv_file, $completed_lines = 0, $lines_per_batch = 30 ) {
 		$this->file = new SplFileObject( $csv_file );
 		$this->file->setFlags( SplFileObject::READ_CSV );
+		$this->set_delimiter();
 
+		$this->file->seek( PHP_INT_MAX );
+		$this->total_lines     = $this->file->key();
+		$this->completed_lines = $completed_lines;
+		$this->is_completed    = $this->completed_lines >= $this->total_lines;
+		$this->lines_per_batch = $lines_per_batch;
+	}
+
+	/**
+	 * Set the delimiter to read the CSV file.
+	 *
+	 * The delimiter detection works testing the delimiter which find more columns.
+	 */
+	private function set_delimiter() {
 		/**
-		 * Change the default CSV delimiter.
+		 * Filters the default CSV delimiter.
 		 *
 		 * @since 3.2.0
+		 * @since 3.3.0 Updated the default to `false`, so it'll get through the delimiter detection.
+		 * @hook sensei_import_csv_delimiter
 		 *
-		 * @param string  $delimiter The CSV file delimiter.
+		 * @param {string} $delimiter The CSV file delimiter.
 		 */
-		$this->file->setCsvControl( apply_filters( 'sensei_import_csv_delimiter', ',' ) );
+		$forced_delimiter = apply_filters( 'sensei_import_csv_delimiter', false );
 
-		$this->file->seek( PHP_INT_MAX );
-		$this->total_lines     = $this->file->key();
-		$this->completed_lines = $completed_lines;
-		$this->is_completed    = $this->completed_lines >= $this->total_lines;
-		$this->lines_per_batch = $lines_per_batch;
+		if ( $forced_delimiter ) {
+			$this->file->setCsvControl( $forced_delimiter );
```

Comment:
We should probably return early here.

Reply:
Good catch! Fixed here changing the commit: https://github.com/Automattic/sensei/pull/3364/commits/95e687bcd3efb99818f8359672461e7ced17e4ff